openedx/api.py

```diff
@@ -1085,6 +1085,14 @@ def enroll_in_edx_course_runs(
                             force_enrollment=force_enrollment,
                         )
                         results.append(enrollment)
+                    else:
+                        raise UnknownEdxApiEnrollException(  # noqa: TRY301
```

Comment:
I feel like this code is getting really complicated, I think we might be better off moving the call to `repair_faulty_edx_user` to earlier in this loop before we call `create_student_enrollment`, because that function will either be a no-op for users that exist or create the user and we don't have to try to make enrollment API requests to find out. That means that you can move this raising of the error earlier and remove the extra `create_student_enrollment` call a few lines above in this `except HTTPError` block. Everything should be a lot less complicated then, and we'll do less API requests, particularly for ones that we know will just fail.